docs/library/openamp.rst

```diff
@@ -0,0 +1,104 @@
+:mod:`openamp` -- provides standard Asymmetric Multiprocessing (AMP) support
+============================================================================
+
+.. module:: openamp
+   :synopsis: provides standard Asymmetric Multiprocessing (AMP) support
+
+The openamp module provides a standard inter-processor communications infrastructure
+for MicroPython. The module handles all of the details of OpenAMP, such as setting up
+the shared resource table, initializing vrings etc.. and provides an API for using the
+RPMsg bus infrastructure with the `Endpoint` class. Furthermore, the module provides
+processor Life Cycle Management (LCM) support, such as loading firmware, starting and
+stopping a remote core, via the `RemoteProc` class.
+
+Example usage::
+
+    import openamp
+    
+    def ept_recv_callback(src_addr, data):
+        print("Received message on endpoint", data)
+    
+    # Create a new RPMsg endpoint to communicate with the remote core.
+    ept = openamp.Endpoint("vuart-channel", callback=ept_recv_callback)
+    
+    # Create a remoteproc object, load its firmware and start it.
+    rproc = openamp.RemoteProc("virtual_uart.elf") # Or entry point address (ex 0x081E0000)
+    rproc.start()
+    
+    while True:
+        if ept.is_ready():
+            ept.send("data")
+
+Functions
+---------
+
+.. function:: ns_callback(ns_callback)
```

Comment:
I think this function should be named differently, at the very least to include the words "new service" in it, because "ns" doesn't mean much to someone new to OpenAMP (or is it "name service"?).

Maybe `new_service_callback()`?

Or `callbacks(new_service_cb)` which leaves open the possibility for additional global-scope callbacks in the future (probably there won't be any, so maybe that's over engineering).

Alternatively, we could follow the `sys` module with things like `sys.displayhook` an make this an attribute:
```py
openamp.new_service_hook = my_function
```

I think that last option (an attribute) is quite a good option.

Reply:
I'm not sure, but I leaning towards the familiar way of setting callbacks, which is `foo.bar_callback(cb)`.